### PR TITLE
transactions: better transactions history display.

### PR DESCRIPTION
**Pull request by zannkukai**
Improves the item transaction history display to highligh transactions
links (all transactions related to the same loan) and improves look and
feel.

Co-authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## How to test?

image

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts [hunk 1]
```diff
@@ -25,53 +27,53 @@ import { marker as _ } from '@biesbjerg/ngx-translate-extract-marker';
...
-
   /** Event on dialog close */
   dialogClose$ = new BehaviorSubject(false);
 
+  // GETTER & SETTER ==========================================================
   /**
-   * Show more link is visible
-   * @return boolean
+   * Is the "show more" link should be visible
+   * @return: True if the link must be visible, False otherwise
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts:56]: Remove `:`

File: projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts [hunk 2]
```diff
@@ -25,53 +27,53 @@ import { marker as _ } from '@biesbjerg/ngx-translate-extract-marker';
...
   get isLinkShowMore(): boolean {
     return this.recordTotals > 0
       && ((this.page * this.itemsPerPage) < this.recordTotals);
   }
 
   /**
-   * Hidden holdings count
-   * @return string
+   * Get the label to use for the hidden transaction logs string.
+   * @return: the translated string to display
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts:65]: Remove `:`

File: projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts [hunk 3]
```diff
@@ -105,13 +108,44 @@ export class CirculationLogsComponent implements OnInit {
       });
   }
 
+  /**
+   * This function will check if two date are in the same time period
+   * @param date1: the first date to check.
+   * @param date2: the second date to check.
+   * @param period: the time period to use to compare both dates
+   * @return: True if dates are similar, False otherwise
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts:116]: Remove `:`

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/record/circulation-logs/circulation-log/circulation-log.component.ts [hunk 4]
```diff
@@ -19,17 +19,28 @@ import { Component, EventEmitter, Input, Output } from '@angular/core';
...
   /** Event for close dialog */
   @Output() closeDialogEvent  = new EventEmitter();
 
   /** Circulation informations is collapsed */
   isCollapsed = true;
+  /** debugMode */
+  debugMode = false;
+
+  get canUseDebugMode(): boolean {
+    return true;
```
Comment (jma) [inline projects/admin/src/app/record/circulation-logs/circulation-log/circulation-log.component.ts:41]: Is it really usefull?

File: projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts [hunk 5]
```diff
@@ -105,13 +108,44 @@ export class CirculationLogsComponent implements OnInit {
...
+      return transDate1.isSame(transDate2, period);
+    }
+    return true;
+  }
+
+  /**
+   * Handler for mouse event
+   * @param record: the record on which the event is triggered.
+   */
+  mouseEventTransaction(record: any): void {
```
Comment (jma) [inline projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts:131]: Do you need to filter by mouse event?

**Review by jma — APPROVED** [review]

File: projects/admin/src/app/record/circulation-logs/circulation-log/circulation-log.component.ts (hunk 4 above)
Comment (zannkukai) [reply projects/admin/src/app/record/circulation-logs/circulation-log/circulation-log.component.ts:41]: You're right. I forget to correctly implement this getter ;-)

File: projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts (hunk 5 above)
Comment (zannkukai) [reply projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts:131]: In a first implementation, I was listening about 'onmouveenter' and 'onmouseleave'.
After a small refactoring, this function is now only listening 'onmouseenter'. But maybe, a future day, this function could listen other events.

**Review by benerken — APPROVED** [review]

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**